comp/core/secrets/impl/secrets.go

```diff
@@ -305,40 +313,56 @@ func (r *secretResolver) Configure(params secrets.ConfigParams) {
 	r.scopeIntegrationToNamespace = params.ScopeIntegrationToNamespace
 	r.allowedNamespace = params.AllowedNamespace
 	r.imageToHandle = params.ImageToHandle
+
+	r.apiKeyFailureRefreshInterval = time.Duration(params.APIKeyFailureRefreshInterval) * time.Minute
 }
 
 func (r *secretResolver) startRefreshRoutine(rd *rand.Rand) {
-	if r.ticker != nil || r.refreshInterval == 0 {
+	if r.ticker != nil {
 		return
 	}
 
-	if r.refreshIntervalScatter {
-		var int63 int64
-		if rd == nil {
-			int63 = rand.Int63n(int64(r.refreshInterval))
+	var timer <-chan time.Time
+	if r.refreshInterval > 0 {
```

Comment:
Since `startRefreshRoutine` is changing purpose, it makes sense to split it up a bit. The functionality related to `refreshInterval`, such as deciding on the scatter amount and the creation of the timer, should move into another method. Let's call it `beginRefreshInterval` or something along those lines. Then the goal of `startRefreshRoutine` is just to launch the goroutine, receive messages, and dispatch them as needed.